XLS-0081d-permissioned-dex/README.md

```diff
@@ -0,0 +1,637 @@
+
+## 4. Transaction: `OfferCreate`
+
+The `OfferCreate` transaction creates an offer on the CLOB DEX. This transaction type already exists on the XRPL, but is being extended as a part of this spec to also support permissioned DEX domains.
+
+### 4.1. Fields
+
+<details>
+<summary>
+
+As a reference, [here](https://xrpl.org/docs/references/protocol/transactions/types/offercreate) are the existing fields for the `OfferCreate` transaction.
+
+</summary>
+
+| Field Name      | Required? | JSON Type | Internal Type | Description                                                                        |
+| --------------- | --------- | --------- | ------------- | ---------------------------------------------------------------------------------- |
+| `Expiration`    |           | `number`  | `UInt32`      | Time after which the offer is no longer active, in seconds since the Ripple Epoch. |
+| `OfferSequence` |           | `number`  | `UInt32`      | An offer to delete first, specified in the same way as `OfferCancel`.              |
+| `TakerGets`     | ✔️        | `string`  | `Amount`      | The amount and type of currency being sold.                                        |
+| `TakerPays`     | ✔️        | `string`  | `Amount`      | The amount and type of currency being bought.                                      |
+
+</details>
+
+We propose a new field:
+
+| Field Name | Required? | JSON Type | Internal Type | Description                                  |
+| ---------- | --------- | --------- | ------------- | -------------------------------------------- |
+| `DomainID` |           | `string`  | `Hash256`     | The domain that the offer must be a part of. |
+
+#### 4.1.1. `Flags`
+
+This spec proposes an additional flag, `tfHybrid` , to indicate whether the offer should consider only the `DomainID`'s Permissioned DEX or also include the open DEX.
+
+> | Flag Name  |  Flag Value  | Description                                                                                   |
+> | :--------: | :----------: | :-------------------------------------------------------------------------------------------- |
+> | `tfHybrid` | `0x00100000` | Indicates the offer is hybrid. This flag cannot be set if the offer doesn't have a `DomainID` |
+
+### 4.2. Failure Conditions
+
+The existing set of failure conditions for `OfferCreate` will continue to exist.
+
+The following will be added:
+
+- The domain doesn't exist.
+- The domain offer creater is neither part of the domain or the domain owner.
+- A hybrid offer with flag `tfHybrid` is attempted to be created without specifying the `DomainID`.
+
+### 4.3. State Changes
+
+The existing set of state changes for `OfferCreate` will continue to exist.
+
+If the `DomainID` is included in the `OfferCreate` transaction, the DEX will use the domain-specific orderbook to process the offer - this applies to hybrid offers as well. Hybrid offers will use domain-specific order book by default for offer crossing.
```

Comment:
This wording is confusing - what do you mean by default? Shouldn't it be using both?

Reply:
This only applies to offer crossing, which reuses payment engine. Currently when a hybrid offer is created, it will only try to cross the domain offers. See [CreateOffer](https://github.com/XRPLF/rippled/blob/053e1af7ff4886d84a278317cbdfcdab1ecd557f/src/xrpld/app/tx/detail/CreateOffer.cpp#L826). The code will call `flow` with the domain. If we want it consume the open book as well, we can perhaps add it through another amendment.

Reply:
That's actually pretty confusing/unexpected behavior IMO... I'd expect it to check both orderbooks even initially.

How is this?
```suggestion
If the `DomainID` is included in the `OfferCreate` transaction, the DEX will use the domain-specific orderbook to process the offer. This applies to hybrid offers as well - hybrid offers will only use the domain-specific order book by default for offer crossing in the execution of that transaction (it can be crossed by open offers later).
```